File: src/ros2_bridge/mappers/sensor_msgs/nav_sat_status.rs
//! Typed mapper for `sensor_msgs/msg/NavSatStatus`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn nav_sat_status_to_bus(msg: ros_env::sensor_msgs::msg::NavSatStatus) -> crate::sensor_msgs::msg::v1::NavSatStatus {
    crate::sensor_msgs::msg::v1::NavSatStatus {
        status: i32::from(msg.status),
        service: u32::from(msg.service),
    }
}

pub(crate) fn nav_sat_status_to_ros(bus: crate::sensor_msgs::msg::v1::NavSatStatus) -> ros_env::sensor_msgs::msg::NavSatStatus {
    ros_env::sensor_msgs::msg::NavSatStatus {
        status: bus.status as i8,
        service: bus.service as u16,
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct SensorMsgsNavSatStatusMapper;

impl TypedTopicMapper for SensorMsgsNavSatStatusMapper {
    type Ros = ros_env::sensor_msgs::msg::NavSatStatus;
    type Bus = crate::sensor_msgs::msg::v1::NavSatStatus;

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(nav_sat_status_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(nav_sat_status_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/sensor_msgs/nav_sat_status.py
"""Generated mapper for `sensor_msgs/msg/NavSatStatus`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.sensor_msgs.msg.v1 import NavSatStatus as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def nav_sat_status_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.status = int(msg.status)
    bus.service = int(msg.service)
    return bus


def nav_sat_status_to_ros(bus):
    from sensor_msgs.msg import NavSatStatus as RosMsg

    out = RosMsg()
    out.status = int(bus.status)
    out.service = int(bus.service)
    return out


class SensorMsgsNavSatStatusMapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from sensor_msgs.msg import NavSatStatus as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return nav_sat_status_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return nav_sat_status_to_ros(bus)
